pkg/controller/certificatemanager/controller.go

```diff
@@ -0,0 +1,308 @@
+	awsconfig, err := conn.awsConfigFn(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := conn.newClientFn(awsconfig)
+	if err != nil {
+		return nil, errors.Wrap(err, errClient)
+	}
+	return &external{c, conn.client}, nil
+}
+
+type external struct {
+	client acmpca.Client
+	kube   client.Client
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.CertificateAuthority)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	if cr.Status.AtProvider.CertificateAuthorityArn == "" {
+		return managed.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	response, err := e.client.DescribeCertificateAuthorityRequest(&awsacmpca.DescribeCertificateAuthorityInput{
+		CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errGet)
+	}
+
+	if response.CertificateAuthority == nil {
+		return managed.ExternalObservation{}, errors.New(errSDK)
+	}
+
+	certificateAuthority := *response.CertificateAuthority
+	current := cr.Spec.ForProvider.DeepCopy()
+	acmpca.LateInitializeCertificateAuthority(&cr.Spec.ForProvider, &certificateAuthority)
+
+	if !cmp.Equal(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{}, errors.Wrap(err, errKubeUpdateFailed)
+		}
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	tags, err := e.client.ListTagsRequest(&awsacmpca.ListTagsInput{
+		CertificateAuthorityArn: aws.String(cr.Status.AtProvider.CertificateAuthorityArn),
+	}).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errListTagsFailed)
+	}
+
+	upToDate := acmpca.IsCertificateAuthorityUpToDate(cr, certificateAuthority, tags.Tags)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errUpToDateFailed)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: upToDate,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+
+	cr, ok := mgd.(*v1alpha1.CertificateAuthority)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	response, err := e.client.CreateCertificateAuthorityRequest(acmpca.GenerateCreateCertificateAuthorityInput(&cr.Spec.ForProvider)).Send(ctx)
+
+	if response != nil {
+
+		cr.Status.AtProvider.CertificateAuthorityArn = aws.StringValue(response.CreateCertificateAuthorityOutput.CertificateAuthorityArn)
+		cr.Status.AtProvider.RenewalPermission = cr.Spec.ForProvider.CertificateRenewalPermissionAllow
+
+		if cr.Spec.ForProvider.CertificateRenewalPermissionAllow {
+
+			_, err = e.client.CreatePermissionRequest(&awsacmpca.CreatePermissionInput{
```

Comment:
I am not sure if this should be part of the `CerftificateAuthority` reconciliation as opposed to another type. Can you provide some additional context here?

Reply:
This PR will manage the Private Certificate Authority(PCA) along with CertificateRenewalPermission.
CertificateRenewalPermission is a part of PCA and can be implemented with it. Input required is only one boolean field i.e. CertificateRenewalPermissionAllow.
ACM PCA package provide separate APIs to create and delete the CertificateRenewalPermission and can be handled with PCA.